pkg/controller/namespace/namespace_controller.go

```diff
@@ -96,35 +106,23 @@ func (r *ReconcileNamespaces) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, fmt.Errorf("failed to generate script: %w", err)
 	}
 
-	var cfg corev1.Secret
-
 	// The default cache-based Client doesn't support cross-namespace queries, unless configured to do so in Manager
 	// Options. However, this is our only use-case for it, so using the non-cached Client instead.
-
-	err = r.apiReader.Get(ctx, client.ObjectKey{Name: webhook.SecretConfigName, Namespace: targetNS}, &cfg)
-	if errors.IsNotFound(err) {
-		log.Info("Creating OneAgent config secret")
-		if err := r.client.Create(ctx, &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      webhook.SecretConfigName,
-				Namespace: targetNS,
-			},
-			Data: data,
-		}); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to create config Secret: %w", err)
-		}
-		return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
-	}
-
+	err = utils.CreateOrUpdateSecretIfNotExists(r.client, r.apiReader, webhook.SecretConfigName, targetNS, data, corev1.SecretTypeOpaque, log)
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to query for config Secret: %w", err)
+		return reconcile.Result{}, err
 	}
 
-	if !reflect.DeepEqual(data, cfg.Data) {
-		log.Info("Updating OneAgent config secret")
-		cfg.Data = data
-		if err := r.client.Update(ctx, &cfg); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to update config Secret: %w", err)
+	imageAnnotation := utils.GetField(apm.Annotations, webhook.AnnotationImage, "")
```

Comment:
You don't access to the Pod annotations here, only on the webhook. This is querying for annotations for the OneAgentAPM object which don't mean anything.

Reply:
Ping. This annotation won't ever exist.